Configure WinSW and provide `--autostart` arg for `safenode-manager add`

jacderida opened the pull request:
### Description

- 0c7376aa **feat: configure winsw in `node-launchpad`**

  The node launchpad will attempt to detect if `winsw.exe` is available, and if it isn't, it will
  download the binary to the application's data directory, and use the `WINSW_PATH` variable to set
  the location. This removes the need for separate steps to download the binary and make it available
  somewhere on `Path`.

  There's also a refactor in here: two functions, `get_config_dir` and `get_launchpad_data_dir_path`
  are moved to the `config` module. It seems like they are both specifically related to configuration
  and it makes sense for them to be in the `config` module rather than `utils`.

- c6253b44 **feat: configure winsw in node manager**

  The node manager will configure `WinSW` either by checking that `safeup` already downloaded it and
  put it alongside the other `safeup`-managed binaries, or by downloading it to the node manager's own
  directory. Please see the comment in the diff for a further explanation here.

- c26516e4 **feat: provide `--autostart` flag for `add` command**

  By default, services will *not* restart upon an OS reboot. If users want that behaviour, they will
  need to explicitly opt in, and that can be on a per-service basis.

  The flag will also be retained on upgrades.

### Type of Change

Please mark the types of changes made in this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

Please ensure all of the following tasks have been completed:

- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have followed the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines for commit messages.

joshuef commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -177,9 +187,273 @@ mod tests {
+        verify_spent_proof_share(key_image, tx, &client).await
+    }
+
+    // Skip spentbook_spend_client_message_should_replicate_to_adults_and_send_ack for now.
```
Let's remove this commented code

joshuef commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -177,9 +187,273 @@ mod tests {
+        }
+    }
+
+    // Skip spentbook_spend_transaction_with_no_inputs_should_return_spentbook_error as this cannot be tested
```
More `//` for chopping block

joshuef commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -177,9 +187,273 @@ mod tests {
+        }
+    }
+
+    // Skip spentbook_spend_with_updated_network_knowledge_should_update_the_node as this should be an sn_node unit test.
```
and here

joshuef commented on sn_client/src/api/spentbook_apis.rs:
```diff
@@ -177,9 +187,273 @@ mod tests {
+    // Skip spentbook_spend_with_updated_network_knowledge_should_update_the_node as this should be an sn_node unit test.
+    // This just needs the state of a node ready to accept the state from the msg.
+    // #[tokio::test(flavor = "multi_thread")]
+    // async fn spentbook_spend_with_updated_network_knowledge_should_update_the_node() -> Result<()> {
```
and here

joshuef reviewed (changes requested):
Some comments to cleanup, but otherwise looking gut

joshuef commented:
bors r+

jacderida committed:
feat: configure winsw in node manager

The node manager will configure `WinSW` either by checking that `safeup` already downloaded it and
put it alongside the other `safeup`-managed binaries, or by downloading it to the node manager's own
directory. Please see the comment in the diff for a further explanation here.

jacderida committed:
chore: use new version of `service-manager` crate

jacderida committed:
feat: configure winsw in `node-launchpad`

The node launchpad will attempt to detect if `winsw.exe` is available, and if it isn't, it will
download the binary to the application's data directory, and use the `WINSW_PATH` variable to set
the location. This removes the need for separate steps to download the binary and make it available
somewhere on `Path`.

There's also a refactor in here: two functions, `get_config_dir` and `get_launchpad_data_dir_path`
are moved to the `config` module. It seems like they are both specifically related to configuration
and it makes sense for them to be in the `config` module rather than `utils`.

jacderida committed:
feat: provide `--autostart` flag for `add` command

By default, services will *not* restart upon an OS reboot. If users want that behaviour, they will
need to explicitly opt in, and that can be on a per-service basis.

The flag will also be retained on upgrades.

Also update `sn-releases` references to the new published version.
